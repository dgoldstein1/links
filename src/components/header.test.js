import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Header from "./header";

const mockStore = configureStore([]);

describe("Header", () => {
  const defaultState = {
    graph: {
      layout: "cluster",
      rootNode: {
        id: 1,
        label: "root node"
      },
      targetNode: {
        id: 2,
        label: "target node"
      }
    }
  };
  describe("matches snapshots", () => {
    let testTable = [
      {
        name: "cluster",
        state: defaultState
      },
      {
        name: "hierachy",
        state: {
          ...defaultState,
          graph: {
            ...defaultState.graph,
            layout: "hierarchy"
          }
        }
      }
    ];

    testTable.forEach(t => {
      it(t.name, () => {
        let store = mockStore(t.state);
        let wrapper = renderer.create(
          <Provider store={store}>
            <Header />
          </Provider>
        );
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
  describe("toggles layout", () => {
    let store = mockStore(defaultState);
    let wrapper = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    // emulate click on button
    let t;
    renderer.act(() => {
      t = wrapper.root
        .findByProps({ id: "layout-toggle-button" })
        .props.onClick();
    });
    expect(t).not.toBeUndefined();
    expect(t).toMatchSnapshot();
  });
  describe("sets settings view", () => {
    let store = mockStore(defaultState);
    let wrapper = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    // emulate click on button
    let t;
    renderer.act(() => {
      t = wrapper.root
        .findByProps({ id: "settings-icon-button" })
        .props.onClick();
    });
    expect(t).not.toBeUndefined();
    expect(t).toMatchSnapshot();
  });
  describe("sets help view", () => {
    let store = mockStore(defaultState);
    let wrapper = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    // emulate click on button
    let t;
    renderer.act(() => {
      t = wrapper.root.findByProps({ id: "help-icon-button" }).props.onClick();
    });
    expect(t).not.toBeUndefined();
    expect(t).toMatchSnapshot();
  });
  describe("sets start path", () => {
    let store = mockStore(defaultState);
    let wrapper = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    // emulate click on button
    renderer.act(() => {
      wrapper.root.findByProps({ id: "search-button" }).props.onClick();
    });
  });
});
