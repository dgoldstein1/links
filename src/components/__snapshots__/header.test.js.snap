// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "layout": "hierarchy",
  "type": "SET_GRAPH_LAYOUT",
}
`;

exports[` 2`] = `
Object {
  "type": "UPDATE_VIEW",
  "view": "settings",
}
`;

exports[` 3`] = `
Object {
  "type": "UPDATE_VIEW",
  "view": "about",
}
`;

exports[`Header matches snapshots cluster 1`] = `
<header
  className="header twoSearchBars sticky"
>
  <a
    className="floatLeft button"
    href="/"
  >
    <img
      alt="logo"
      className="logo"
      src="title.png"
    />
  </a>
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <input
      aria-autocomplete="list"
      aria-expanded={false}
      autoComplete="off"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="starting at.."
      role="combobox"
      value="root node"
    />
  </div>
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <input
      aria-autocomplete="list"
      aria-expanded={false}
      autoComplete="off"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="ending at.."
      role="combobox"
      value="target node"
    />
  </div>
  <button
    id="search-button"
    onClick={[Function]}
  >
    <span
      className="icon-search"
    />
    search
  </button>
  <button
    id="layout-toggle-button"
    onClick={[Function]}
  >
    hierarchy
  </button>
  <button
    className="floatRight"
    id="settings-icon-button"
    onClick={[Function]}
  >
    <span
      className="icon-settings"
    />
  </button>
  <button
    className="floatRight"
    id="help-icon-button"
    onClick={[Function]}
  >
    <span
      className="icon-help"
    />
  </button>
</header>
`;

exports[`Header matches snapshots hierachy 1`] = `
<header
  className="header twoSearchBars sticky"
>
  <a
    className="floatLeft button"
    href="/"
  >
    <img
      alt="logo"
      className="logo"
      src="title.png"
    />
  </a>
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <input
      aria-autocomplete="list"
      aria-expanded={false}
      autoComplete="off"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="starting at.."
      role="combobox"
      value="root node"
    />
  </div>
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <input
      aria-autocomplete="list"
      aria-expanded={false}
      autoComplete="off"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="ending at.."
      role="combobox"
      value="target node"
    />
  </div>
  <button
    id="search-button"
    onClick={[Function]}
  >
    <span
      className="icon-search"
    />
    search
  </button>
  <button
    id="layout-toggle-button"
    onClick={[Function]}
  >
    cluster
  </button>
  <button
    className="floatRight"
    id="settings-icon-button"
    onClick={[Function]}
  >
    <span
      className="icon-settings"
    />
  </button>
  <button
    className="floatRight"
    id="help-icon-button"
    onClick={[Function]}
  >
    <span
      className="icon-help"
    />
  </button>
</header>
`;
